hyp3_sdk/jobs.py

```diff
@@ -1,96 +1,221 @@
+            files: Optional[List] = None,
+            browse_images: Optional[List] = None,
+            thumbnail_images: Optional[List] = None,
+            expiration_time: Optional[datetime] = None
+    ):
+        self.job_id = job_id
         self.job_type = job_type
+        self.request_time = request_time
+        self.status_code = status_code
+        self.user_id = user_id
+        self.name = name
         self.job_parameters = job_parameters
-
-    def to_dict(self) -> dict:
-        """
-        Returns:
-            A dictionary representation of the Job object
-        """
-        return {
-            'job_parameters': {
-                **self.job_parameters
-            },
+        self.files = files
+        self.browse_images = browse_images
+        self.thumbnail_images = thumbnail_images
+        self.expiration_time = expiration_time
+
+    def __repr__(self):
+        return str(self.to_dict())
+
+    def __eq__(self, other):
+        return self.__dict__ == other.__dict__
+
+    @staticmethod
+    def from_dict(input_dict: dict):
+        expiration_time = parse_date(input_dict['expiration_time']) if input_dict.get('expiration_time') else None
+        return Job(
+            job_type=input_dict['job_type'],
+            job_id=input_dict['job_id'],
+            request_time=input_dict['request_time'],
+            status_code=input_dict['status_code'],
+            user_id=input_dict['user_id'],
+            name=input_dict.get('name'),
+            job_parameters=input_dict.get('job_parameters'),
+            files=input_dict.get('files'),
+            browse_images=input_dict.get('browse_images'),
+            thumbnail_images=input_dict.get('thumbnail_images'),
+            expiration_time=expiration_time
+        )
+
+    def to_dict(self, for_resubmit: bool = False):
+        job_dict = {
             'job_type': self.job_type,
-            'name': self.job_name,
         }
 
+        for key in ['job_id', 'request_time', 'status_code', 'user_id']:
+            value = self.__getattribute__(key)
+            if not for_resubmit:
+                job_dict[key] = value
+
+        for key in ['name', 'job_parameters']:
+            value = self.__getattribute__(key)
+            if value is not None:
+                job_dict[key] = value
 
-def make_job(job_type: str, job_name: str, job_parameters: dict = {}) -> Job:
-    """Make a generic Job object
+        for key in ['files', 'browse_images', 'thumbnail_images']:
+            value = self.__getattribute__(key)
+            if value is not None and not for_resubmit:
+                job_dict[key] = value
 
-    Args:
-        job_type: The job type
-        job_name: A name for the job (must be <= 20 characters)
-        job_parameters: Extra job parameters specifying custom processing options
+        if self.expiration_time is not None and not for_resubmit:
+            job_dict['expiration_time'] = self.expiration_time.isoformat(timespec='seconds')
```

Comment:
should have a single `if for_resubmit:` that wraps the three associated paragraphs (everything but type/name/parameters)